File: API/BookingMaster/BookingMaster/Data/Model/Accommodation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using static BookingMaster.Data.Model.Enums;

namespace BookingMaster.Data.Model
{
    public class Accommodation
    {
        [Required]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        public string Name { get; set; }
        public string Phone_number { get; set; }
        public AccomodationType Type { get; set; }
        public string Location { get; set; }
        public int Total_Number_Of_Rooms { get; set; }
        public int Number_Of_Free_Rooms { get; set; }
        public bool IsPublished { get; set; }

        public ICollection<Room> Rooms { get; set; }
    }
}

File: BookingMaster/BookingMaster/app/src/main/java/com/example/bookingmaster/model/Accommodation.kt
package com.example.bookingmaster.model

import com.squareup.moshi.JsonClass

/** Get products - by location **/
@JsonClass(generateAdapter = true)
data class AccommodationResponse(val accomodations: List<Accommodation>)

@JsonClass(generateAdapter = true)
data class Accommodation(
    val id: Int,
    var name: String,
    var phone_number: String,
    val type: Int,
    val location: String,
    var total_Number_Of_Rooms: Int,
    var isPublished: Boolean,
    val rooms: List<Room>
)

@JsonClass(generateAdapter = true)
data class Room(
    val id: Int,
    var price: Int,
    var currency: String,
    var capacity: Int,
    val accommodationId: Int,
)